### PR TITLE
Improve shape of the Control Flow Graph when throw expression is used in conditional context.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@333fred, @mavasani, @cston, @dotnet/roslyn-compiler, @dotnet/analyzer-ioperation Please review

**Review by gafter — APPROVED** [review]
:shipit:

**Review by mavasani — APPROVED** [review]

**Comment by AlekseyTs** [comment]
@333fred, @cston, @dotnet/roslyn-compiler Please review, need a second review

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (333fred) [inline src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: Typo: specialy -> specially

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (333fred) [inline src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: Do we know what node this originally corresponds to, and do we have a bug tracking supporting it natively?

**Review by 333fred — APPROVED** [review]
LGTM with bug for tracking the NoneOperation (iteration 1)

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs
Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: > Do we know what node this originally corresponds to, and do we have a bug tracking supporting it natively?

This is not a bug, but by design behavior. It is created by the CFG builder for throw expression. This was discussed at design meeting. I believe that after this change it will be exposed only for error scenarios, i.e. erroneous use of a throw expression.

---
In reply to: [193964473](https://github.com/dotnet/roslyn/pull/27612#discussion_r193964473) [](ancestors = 193964473)

**Pull request merged**
